src/intervals/intervalstream.jl

```diff
@@ -0,0 +1,180 @@
+
+
+# IntervalStreams
+# ---------------
+#
+# Often we'd rather avoid reading a large interval dataset into an
+# IntervalCollection. We can still do efficient intersections if we can assume
+# the data is sorted
+
+
+typealias IntervalStreamOrArray{T} Union(AbstractArray{Interval{T}}, IntervalStream{T})
+
+type IntervalStreamIntersectIterator{S, T}
+    a::IntervalStreamOrArray{S}
+    b::IntervalStreamOrArray{T}
+    seqname_isless::Function
+
+    # buffered intervals from a
+    a_buffer::StreamBuffer{Interval{S}}
+    b_interval::Interval{T}
+
+    function IntervalStreamIntersectIterator(a::IntervalStreamOrArray{S},
+                                             b::IntervalStreamOrArray{T},
+                                             seqname_isless::Function)
+        return new(a, b, seqname_isless, StreamBuffer{Interval{S}}())
+    end
+end
+
+
+immutable IntervalStreamIntersectIteratorState{U, V}
+    a_buffer_pos::Int
+    a_state::U
+    b_state::V
+end
+
+
+@doc """
+Intersect two `IntervalStreams` returning an iterator over all pairs of
+intersecting intervals.
+""" ->
+function Base.intersect{S, T}(a::IntervalStreamOrArray{S},
```

Comment:
This generic function is imported explicit on this module, therefore we don't need `Base.`